internal/provider/resource_cloudflare_ruleset_test.go

```diff
@@ -2674,3 +2734,72 @@ func testAccCheckCloudflareRulesetActionParametersOverridesThrashingStatus(rnd,
     }
   }`, rnd, zoneID, zoneName, status)
 }
+
+func testAccCloudflareRulesetCacheSettings(rnd, accountID, zoneID string) string {
```

Comment:
we will probably need to split this into multiple tests to exercise all conditions without conflicts. this currently errors

```
TF_ACC=1 go test $(go list ./...) -v -run "^TestAccCloudflareRuleset_CacheSettings" -count 1 -parallel 1 -timeout 120m -parallel 1
?   	github.com/cloudflare/terraform-provider-cloudflare	[no test files]
=== RUN   TestAccCloudflareRuleset_CacheSettings
=== PAUSE TestAccCloudflareRuleset_CacheSettings
=== CONT  TestAccCloudflareRuleset_CacheSettings
    resource_cloudflare_ruleset_test.go:1560: Step 1/1 error: Error running apply: exit status 1

        Error: error creating ruleset my basic cache settings ruleset: cannot contain both status_code and status_code_range, default is useless in respect_origin mode, in custom key use user/device_type field instead of cache_by_device_type

          with cloudflare_ruleset.xtakiydsrb,
          on terraform_plugin_test.tf line 2, in resource "cloudflare_ruleset" "xtakiydsrb":
           2: 	resource "cloudflare_ruleset" "xtakiydsrb"{

--- FAIL: TestAccCloudflareRuleset_CacheSettings (6.93s)
FAIL
FAIL	github.com/cloudflare/terraform-provider-cloudflare/internal/provider	7.333s
?   	github.com/cloudflare/terraform-provider-cloudflare/tools/cmd/changelog-check	[no test files]
?   	github.com/cloudflare/terraform-provider-cloudflare/tools/cmd/maintainer-only-file-check	[no test files]
?   	github.com/cloudflare/terraform-provider-cloudflare/tools/cmd/tf-log-check	[no test files]
?   	github.com/cloudflare/terraform-provider-cloudflare/version	[no test files]
FAIL
make: *** [testacc] Error 1
```

Reply:
Yes, debugged this test (now it's working) and add more tests